enhancements/olm/safe-cluster-upgrades.md

```diff
@@ -0,0 +1,265 @@
+
+```yaml=
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    # Prevent upgrades to OpenShift Version 4.9
+    operators.coreos.com/maxOpenShiftVersion: 4.8.0 
+```
+
+> NOTE: OLM should allow users to specify Major.Minor.Patch versions in this field BUT OLM can only prevent Major.Minor OpenShift upgrades. By allowing Major.Minor.Patch versions to be specified, OLM will not need to modify the existing annotations if OpenShift ever allows ClusterOperators to prevent patch upgrades.
+
+Assume that the CSV shown above is the only CSV on the Cluster. In this case, OLM would compare the value of the `operators.coreos.com/maxOpenShiftVersion` annotation against the version of the Cluster to decide it's upgradeable status.
+
+Examples can be found below:
+
+| OpenShift Cluster Version   | CSV's Max OpenShift Version | Is Upgradeable?|
+| --------------------------- | --------------------------- | -------------- |
+| 4.6.z                       | 4.8.0                       | True           |
+| 4.7.z                       | 4.8.0                       | True           |
+| 4.8.z                       | 4.8.0                       | False          |
+
+As this is an opt-in feature, there are instances where OLM will be unable to determine if the upgrade is safe:
+
+- The CSV is missing the `operators.coreos.com/maxOpenShiftVersion` annotation
+- The value associated with the `operators.coreos.com/maxOpenShiftVersion` is not a valid Semantic Version (Valid Example: x.y.z).
+
+OLM will report the following upgradeable conditions based on the collection of CSV present on cluster:
+
+| Key                  | Meaning                                                                                            |
+| ---------------------| -------------------------------------------------------------------------------------------------- |
+| Upgradeable    CSV   | A CSV with a valid upgradeable annotation that will run on the next Minor version of OpenShift     |
+| Non-Upgradeable  CSV | A CSV with a valid upgradeable annotation that will not run on the next Minor version of OpenShift |
+| Undeterminable CSV   | A CSV that does not include a valid upgradeable annotation                                         |
+
+| Upgradeable CSV Present | Indeterminate CSV Present | Non-Upgradeable CSV Present | Status          | Message            |
+| ----------------------- | ------------------------- | --------------------------- |---------------- | ------------------ |
+| Yes                     | No                        | No                          | Upgradeable     | Ready for upgrade  |
+| Don't Care              | Yes                       | No                          | Upgradeable     | Warning            |
+| Don't Care              | Don't Care                | Yes                         | Not Upgradeable | Upgrade is blocked |
+
+Ben Parees  11:49 AM|
```

Comment:
cut+paste issue?  :)
also i like the sample messages you provided in slack, so i'd go ahead and put them in.

Reply:
(I was just being lazy when i wrote up the sample table, so i didn't write out the more detailed sample message)

Reply:
Ha yes a copy paste issue! Updated in the latest version of the PR! Please mark as resolved if we're all set!